modules/porous_flow/src/userobjects/PorousFlowDictator.C

```diff
@@ -0,0 +1,120 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+//  Holds maps between PorousFlow variables (porepressure, saturations) and the variable number used by MOOSE.
+#include "PorousFlowDictator.h"
+
+template<>
+InputParameters validParams<PorousFlowDictator>()
+{
+  InputParameters params = validParams<GeneralUserObject>();
+  params.addClassDescription("Holds information on the PorousFlow variable names");
+  params.addRequiredCoupledVar("porous_flow_vars", "List of primary variables that are used in the PorousFlow simulation.  Jacobian entries involving derivatives wrt these variables will be computed.  In single-phase models you will just have one (eg \'pressure\'), in two-phase models you will have two (eg \'p_water p_gas\', or \'p_water s_water\'), etc.");
+  params.addRequiredParam<unsigned int>("number_fluid_phases", "The number of fluid phases in the simulation");
+  params.addRequiredParam<unsigned int>("number_fluid_components", "The number of fluid components in the simulation");
+  return params;
+}
+
+PorousFlowDictator::PorousFlowDictator(const InputParameters & parameters) :
+    GeneralUserObject(parameters),
+    Coupleable(this, false),
+    ZeroInterface(parameters),
+    _num_variables(coupledComponents("porous_flow_vars")),
+    _num_phases(getParam<unsigned int>("number_fluid_phases")),
+    _num_components(getParam<unsigned int>("number_fluid_components"))
+{
+  unsigned int max_moose_var_num_seen = 0;
+
+  _moose_var_num.resize(_num_variables);
+  for (unsigned int i = 0; i < _num_variables; ++i)
+  {
+    _moose_var_num[i] = coupled("porous_flow_vars", i);
+    max_moose_var_num_seen = (max_moose_var_num_seen > _moose_var_num[i] ? max_moose_var_num_seen : _moose_var_num[i]);
+  }
+
+  _pf_var_num.resize(max_moose_var_num_seen + 1);
```

Comment:
The dictator would not eliminate me, it would basically commit suicide by allocating all the physical memory on the computer and then some.